pkg/pillar/cmd/vaultmgr/vaultmgr.go

```diff
@@ -1084,6 +1084,11 @@ func handleVaultKeyFromControllerImpl(ctxArg interface{}, key string,
 		log.Noticef("%s unlocked using key type %s", defaultVault,
 			etpm.CompareLegacyandSealedKey().String())
 
+		// make sure we (re)publish the vault key to Controller
+		if err := publishVaultKey(ctx, types.DefaultVaultName); err != nil {
+			log.Errorf("Failed to publish Vault Key, %v", err)
+		}
+
```

Comment:
Probably I cannot understand something, but few lines below we run the same command. Moreover, inside of `publishVaultKey` we check for `ctx.defaultVaultUnlocked`, so the correct place to run `publishVaultKey` is after setting `ctx.defaultVaultUnlocked = true`, that is, few lines below (as we do it now).

Reply:
I'm the confused one. 
I thought when I read and modified the code that there was a case when it didn't publish the key on boot when the vault was unlocked, but that doesn't seem to be the case.

Thus the only problematic case is if 
 - device is installed
 - device boots the first time; creates vault key and /persist/vault; seals key under PCRs
 - network is out so sending the attestation with the encrypted vault key fails
 - device is powered off
 - device is powered on and boots but the PCRs are different.

In that case there is no way to recover the vault key; only choice is to destroy and recreate the vault.
If the PCRs match then the unseal will happen and vaultmgr will publish.

So what we need is a way for vaultmgr to detect that the remote attestation passed but there was no encrypted vault key received from the controller, and in that case delete+recreate.

Reply:
Implementation is here: https://github.com/lf-edge/eve/pull/2755

Reply:
So which PR should be reviewed ? #2755 or this one ?

Reply:
Both of them. This one improves the logging. 2755 has a proposed fix.